zuul-core/src/main/java/com/netflix/zuul/filters/ZuulFilter.java

```diff
@@ -98,4 +100,14 @@
      * @return
      */
     HttpContent processContentChunk(ZuulMessage zuulMessage, HttpContent chunk);
+
+    /**
+     * Returns the internal profiling tag for this filter.   This method is NOT API stable, and may be removed.
+     * This method is internal.
+     * @return
+     */
+    default Tag perfmarkTag() {
```

Comment:
Doesn't adding this method here, remove the need to add it to all the other interfaces?

Reply:
This method is for the callers (in the core fo Zuul), but it just returns a dummy tag that has no information.    This is just to make it safe to call on any subclasses, incase they dont yet implement this method (i.e. code outside of ours).    The sub classes what to implement this method to return their named tags, which are cached on each filter instance.